engine/src/main/java/org/hibernate/search/util/impl/ReflectionHelper.java

```diff
@@ -115,13 +115,34 @@ public static Object getMemberValue(Object bean, XMember getter) {
 	}
 
 	/**
-	 * Checks whether the specified class contains any Search specific annotations.
+	 * Checks whether the specified class contains any Search-specific annotations, i.e. if
+	 * the class itself or one of its superclasses carries such an annotation.
 	 *
 	 * @param mappedClass the {@code XClass} to check for Search annotations
 	 *
 	 * @return Returns {@code true} if the class contains at least one Search annotation, {@code false} otherwise
 	 */
 	public static boolean containsSearchAnnotations(XClass mappedClass) {
+		List<XClass> hierarchy = ReflectionHelper.createXClassHierarchy( mappedClass );
+
+		for ( XClass clazz : hierarchy ) {
```

Comment:
I'm wondering if it would be worth it to exclude `java.lang.Object`? Sounds like some useless work. Using the programmatic API to declare Search annotations on it would be weird to say the least.

But maybe, it's safer to also take it into account.

WDYT?